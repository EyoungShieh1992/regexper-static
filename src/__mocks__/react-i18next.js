const reactI18next = jest.requireActual('react-i18next');
const i18n = require('i18n');

module.exports = {
  ...reactI18next,
  Trans: require('__mocks__/component-mock').buildMock(reactI18next.Trans),
  useTranslation: () => ({ i18n, t: i18n.mockT })
};
