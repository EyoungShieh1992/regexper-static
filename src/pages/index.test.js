jest.mock('components/Metadata', () =>
  require('__mocks__/component-mock')('components/Metadata'));
jest.mock('components/Message', () =>
  require('__mocks__/component-mock')('components/Message'));
jest.mock('components/App', () =>
  require('__mocks__/component-mock')('components/App'));

import React from 'react';
import { render } from 'react-testing-library';

import IndexPage from 'pages/index';

const queryResult = {
  site: {
    siteMetadata: {
      description: 'Test description',
      defaultSyntax: 'testJs',
      syntaxList: [
        { id: 'testJS', name: 'Testing JS' },
        { id: 'other', name: 'Other' }
      ]
    }
  }
};

describe('Index Page', () => {
  test('rendering', () => {
    const { asFragment } = render(
      <IndexPage location={{ hash: '' }} data={ queryResult } />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('rendering with an expression on the URL', () => {
    const { asFragment } = render(
      <IndexPage location={{
        hash: '#syntax=test&expr=testing',
        href: 'http://example.com'
      }} data={ queryResult } />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
