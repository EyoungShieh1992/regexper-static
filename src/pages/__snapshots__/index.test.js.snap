// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Page rendering 1`] = `
<DocumentFragment>
  <span
    data-component="Metadata"
    data-props="{
  \\"description\\": \\"Test description\\"
}"
  />
  <noscript />
  <span
    data-component="withI18nextTranslation(App)"
    data-props="{
  \\"syntaxList\\": [
    {
      \\"id\\": \\"testJS\\",
      \\"name\\": \\"Testing JS\\"
    },
    {
      \\"id\\": \\"other\\",
      \\"name\\": \\"Other\\"
    }
  ],
  \\"syntax\\": \\"testJs\\",
  \\"expr\\": \\"\\",
  \\"permalinkUrl\\": null
}"
  />
</DocumentFragment>
`;

exports[`Index Page rendering with an expression on the URL 1`] = `
<DocumentFragment>
  <span
    data-component="Metadata"
    data-props="{
  \\"description\\": \\"Test description\\"
}"
  />
  <noscript />
  <span
    data-component="withI18nextTranslation(App)"
    data-props="{
  \\"syntaxList\\": [
    {
      \\"id\\": \\"testJS\\",
      \\"name\\": \\"Testing JS\\"
    },
    {
      \\"id\\": \\"other\\",
      \\"name\\": \\"Other\\"
    }
  ],
  \\"syntax\\": \\"test\\",
  \\"expr\\": \\"testing\\",
  \\"permalinkUrl\\": \\"http://example.com\\"
}"
  />
</DocumentFragment>
`;
